src/modules/iotjs_module_fs.c

```diff
@@ -41,7 +41,61 @@ static void iotjs_fs_reqwrap_destroy(iotjs_fs_reqwrap_t* fs_reqwrap) {
   IOTJS_RELEASE(fs_reqwrap);
 }
 
-iotjs_jval_t MakeStatObject(uv_stat_t* statbuf);
+
+static iotjs_jval_t StatsIsTypeOf(iotjs_jval_t stats, int type) {
+  iotjs_jval_t mode = iotjs_jval_get_property(stats, IOTJS_MAGIC_STRING_MODE);
+
+  int mode_number = (int)iotjs_jval_as_number(mode);
+
+  jerry_release_value(mode);
+
+  return jerry_create_boolean((mode_number & S_IFMT) == type);
+}
+
+
+JS_FUNCTION(StatsIsDirectory) {
+  DJS_CHECK_THIS(object);
+  iotjs_jval_t stats = JS_GET_THIS(object);
+  return StatsIsTypeOf(stats, S_IFDIR);
+}
+
+
+JS_FUNCTION(StatsIsFile) {
+  DJS_CHECK_THIS(object);
+  iotjs_jval_t stats = JS_GET_THIS(object);
+  return StatsIsTypeOf(stats, S_IFREG);
+}
+
+
+static iotjs_jval_t MakeStatObject(uv_stat_t* statbuf) {
+  iotjs_jval_t jstat = iotjs_jval_create_object();
+  iotjs_jval_t stat_prototype = iotjs_jval_create_object();
```

Comment:
With this approach we always recreate the Stat object prototype and each time it is a different one. So we don't get anything with using prototypes as two stat objects prototype will be different thus there isn't any sense to have a prototype object at all.

The old way was to ensure that every Stat object will have the same prototype.